airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -494,9 +494,114 @@ def validate_start_time(self, requested_start_time: int, value: int = 1) -> int:
         return super().validate_start_time(requested_start_time, value=3)
 
 
-class TicketSubstream(HttpSubStream, IncrementalZendeskSupportStream):
+class TicketMetrics(SourceZendeskSupportStream):
 
-    cursor_field = "generated_timestamp"
+    name = "ticket_metrics"
+    cursor_field = "_ab_updated_at"
+
+    def __init__(self, subdomain: str, start_date: str, ignore_pagination: bool = False, **kwargs):
+        super().__init__(subdomain=subdomain, start_date=start_date, ignore_pagination=ignore_pagination, **kwargs)
+        self._state_migrator = TicketMetricsStateMigration()
+        self._parent_stream: SourceZendeskSupportStream = None
```

Comment:
I think the field name `parent_stream` is misleading in term of what it's purpose is for the stream. I typically think of the parent stream to be something like a Tickets, but this is really more of a conditional branch (based on state) of which underlying implementation of the TicketMetrics stream is implemented, not truly a parent like Tickets. The way this is designed StatefulTicketMetrics is a substream and StatelessTicketMetrics is really more of a normal stream that doesn't allow for checkpointing.

And looking how it's used in the code in implementations of read_records(), parse_response() we're just relying on an underlying implementation not a parent. I'm trying to think of a better word for this. Maybe even something generic like `_stream_implementation` or `retrieval_stream` would be less confusing

Reply:
Yeah -- that's fair. The way I was thinking about it is that "parent_stream" kind of took the place of "super" so it's kind of like a parent class to the stream. But agree -- it's confusing naming. I'll update.